cli/azd/internal/appdetect/spring_boot.go

```diff
@@ -268,22 +334,53 @@ func detectStorageAccountAccordingToSpringCloudStreamBinderMavenDependencyAndPro
 			}
 		}
 		if containsInBindingName != "" {
-			containerNamePropertyMap := make(map[string]string)
-			for key, value := range springBootProject.applicationProperties {
-				if strings.HasSuffix(key, targetPropertyName) {
-					containerNamePropertyMap[key] = value
-				}
-			}
-			newDep := AzureDepStorageAccount{
-				ContainerNamePropertyMap: containerNamePropertyMap,
-			}
-			azdProject.AzureDeps = append(azdProject.AzureDeps, newDep)
-			logServiceAddedAccordingToMavenDependencyAndExtraCondition(newDep.ResourceDisplay(), targetGroupId,
-				targetArtifactId, "binding name ["+containsInBindingName+"] contains '-in-'")
-			for property, containerName := range containerNamePropertyMap {
-				log.Printf("  Detected Storage container name: [%s] for [%s] by analyzing property file.",
-					containerName, property)
-			}
+			detectStorageAccountAccordingToProperty(azdProject, springBootProject.applicationProperties,
+				targetGroupId, targetArtifactId, targetPropertyName,
+				"binding name ["+containsInBindingName+"] contains '-in-'")
+		}
+	}
+}
+
+func detectStorageAccountAccordingToSpringIntegrationEventhubsMavenDependencyAndProperty(
+	azdProject *Project, springBootProject *SpringBootProject) {
+	var targetGroupId = "com.azure.spring"
+	var targetArtifactId = "spring-cloud-azure-starter-integration-eventhubs"
+	var targetPropertyName = "spring.cloud.azure.eventhubs.processor.checkpoint-store.container-name"
+	if hasDependency(springBootProject, targetGroupId, targetArtifactId) {
+		detectStorageAccountAccordingToProperty(azdProject, springBootProject.applicationProperties,
+			targetGroupId, targetArtifactId, targetPropertyName, "")
+	}
+}
+
+func detectStorageAccountAccordingToSpringMessagingEventhubsMavenDependencyAndProperty(
+	azdProject *Project, springBootProject *SpringBootProject) {
+	var targetGroupId = "com.azure.spring"
+	var targetArtifactId = "spring-messaging-azure-eventhubs"
+	var targetPropertyName = "spring.cloud.azure.eventhubs.processor.checkpoint-store.container-name"
+	if hasDependency(springBootProject, targetGroupId, targetArtifactId) {
+		detectStorageAccountAccordingToProperty(azdProject, springBootProject.applicationProperties,
+			targetGroupId, targetArtifactId, targetPropertyName, "")
+	}
+}
+
+func detectStorageAccountAccordingToProperty(azdProject *Project, applicationProperties map[string]string,
+	targetGroupId string, targetArtifactId string, targetPropertyName string, extraCondition string) {
+	containerNamePropertyMap := make(map[string]string)
+	for key, value := range applicationProperties {
+		if strings.HasSuffix(key, targetPropertyName) {
+			containerNamePropertyMap[key] = value
+		}
+	}
+	if len(containerNamePropertyMap) > 0 {
+		newDep := AzureDepStorageAccount{
+			ContainerNamePropertyMap: containerNamePropertyMap,
+		}
+		azdProject.AzureDeps = append(azdProject.AzureDeps, newDep)
```

Comment:
Is it possible that one project has the 3 dependencies at the same time? In this case the `append(azdProject.AzureDeps, newDep)` will be executed 3 times. Is this same to your expectation? In my understanding, in this case it should contain 1 event hub service with 3 different event hub name, but current code will try to create 3 event hub services, but 3 event hub services is supported in current SJAD.

Reply:
Yes, if one project has the 3 dependencies, we will detect one by one, so `addAzureDepEventHubsIntoProject` will be executed 3 times. But now, I have updated to achieve that only one event hubs will be added, and will contain multiple `FromDependencies` and multiple event hub names.